timemachine/fe/single_topology_v3.py

```diff
@@ -799,6 +784,48 @@ def __init__(self, mol_a, mol_b, core, forcefield, lambda_angles=0.4, lambda_tor
         self.c_to_a = {v: k for k, v in enumerate(self.a_to_c)}
         self.c_to_b = {v: k for k, v in enumerate(self.b_to_c)}
 
+
+class SingleTopologyV3(AtomMapMixin):
+    def __init__(self, mol_a, mol_b, core, forcefield, lambda_angles=0.4, lambda_torsions=0.7):
+        """
+        SingleTopology combines two molecules through a common core. The combined mol has
+        atom indices laid out such that mol_a is identically mapped to the combined mol indices.
+        The atoms in the mol_b's R-group is then glued on to resulting molecule.
+
+        Parameters
+        ----------
+        mol_a: ROMol
+            First guest
+
+        mol_b: ROMol
+            Second guest
+
+        core: np.array (C, 2)
+            Atom mapping from mol_a to mol_b.
+
+        forcefield: ff.Forcefield
+            Forcefield to be used for parameterization.
+
+        lambda_angles, lambda_torsions: float
+            For ring opening/closing transformations, alchemical parameter values controlling the intervals over which
+            bonds, angles, and torsions are interpolated. Note that these have no effect on terms not involved in ring
+            opening/closing.
+
+            - Bonds are interpolated in the interval [0, lambda_angles]
+            - Angles are interpolated in the interval [lambda_angles, lambda_torsions]
+            - Torsions are interpolated in the interval [lambda_torsions, 1]
+        """
+        # initialize the mixin to get the a_to_c, b_to_c, c_to_a, c_to_b, and c_flags
+        AtomMapMixin.__init__(self, mol_a, mol_b, core)
```

Comment:
I'm curious, is there an advantage to using a mixin here versus modifying the `SingleTopologyV3` constructor to take an `AtomMap` instead of `mol_a`, `mol_b`, `core`? For the uses I'm imagining the latter seems a bit simpler (though not a huge difference)

Reply:
Going to answer this in the other PR #862